force-app/main/default/classes/PMT_RefundController.cls

```diff
@@ -44,34 +44,91 @@ public with sharing class PMT_RefundController {
         }
         set;
     }
+ 
+    @TestVisible
+    private static PMT_RefundService refundService {
+        get {
+            if (refundService == null) {
+                refundService = new PMT_RefundService();
+            }
+            return refundService;
+        }
+        set;
+    }
 
     @AuraEnabled
-    public static RefundView getPermissionData() {
+    public static RefundView getInitialView(Id paymentId) {
         RefundView refundView = new RefundView();
-        refundView.hasRequiredPermissions = hasRequiredPermission();
+        setPermissionData(refundView);
 
+        if (!refundView.hasRequiredPermissions) {
+            return refundView;
+        }
+
+        npe01__OppPayment__c originalPayment = paymentSelector.getPaymentRecordForRefund(paymentId);
+        refundView.paymentDate = originalPayment.npe01__Payment_Date__c;
+        refundView.paymentAmount = originalPayment.npe01__Payment_Amount__c;
+        refundView.currencyCode = (String) originalPayment.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
```

Comment:
Need to check for Multicurrency org here.

Reply:
I test it on this one, `Sobject.get()` will return null on the currency code when multicurrency is disabled and the default will be set to USD in the front end.